### ISSUE TITLE
Sandbox slow to start up for the first time

### ISSUE THREAD

**Issue by heidihoward** [microsoft/CCF#5246]
The sandbox is slow to start up when it's ran for the first time on a new machine. Today, it took ~5 mins to start using 32 core 64GB instance of codespaces. By default, it also doesn't print anything to stdout/err during this time so it's not clear to the patient user what's going on.

On a fresh codespaces instance, I ran:
```
$ cd getting_started/setup_vm/
$ ./run.sh app-dev.yml --extra-vars "platform=virtual" --extra-vars "clang_version=15"
```
and then
```
root@codespaces-92eb25:/workspaces/CCF/getting_started/setup_vm# date; /opt/ccf_virtual/bin/sandbox.sh
Fri May  5 15:20:40 UTC 2023
Setting up Python environment...
No package/app specified. Defaulting to installed JS logging app
Python environment successfully setup
[15:25:28.889] Virtual mode enabled
[15:25:28.889] Starting 1 CCF node...
[15:25:32.185] Started CCF network with the following nodes:
[15:25:32.185]   Node [0] = https://127.0.0.1:8000
[15:25:32.185] You can now issue business transactions to the /opt/ccf_virtual/bin/../lib/libjs_generic application
[15:25:32.186] Loaded JS application: /opt/ccf_virtual/bin/../samples/logging/js
[15:25:32.186] Keys and certificates have been copied to the common folder: /workspaces/CCF/getting_started/setup_vm/workspace/sandbox_common
[15:25:32.186] See https://microsoft.github.io/CCF/main/use_apps/issue_commands.html for more information
[15:25:32.186] Press Ctrl+C to shutdown the network
^C[15:25:38.870] Stopping all CCF nodes...
[15:25:38.871] All CCF nodes stopped.
```

**Comment by achamayou** [comment]
The main reason the sandbox is slow the first time is that it creates a virtual environment with a liberal amount of Python dependencies needed for testing. Not all of them are useful for the sandbox alone, but our test dependency management is not very fine grained as it stands. A lot of things are only really necessary when plotting cimetrics charts or running piccolo for example, not for the sandbox itself. We could improve on that.

We could also leave dependency management up to users (they could choose to install in their global Python environment) rather than create a virtual environment. That would probably save some time in the average human user case, which likely has most of what we need already, but make outliers much worse in case of conflicts or automated environments. I don't think we want to change that.

The lack of feedback is somewhat deliberate: it was thought that letting pip print out all the details of what it's doing would be distracting and off topic. Users may not know, or care, that the sandbox is using Python. Perhaps there is a way to get a progress bar of some sort, or a way to indicate activity that is less detailed that full output. Or perhaps we're taking the wrong side of that tradeoff and just have a more verbose output.

**Comment by achamayou** [comment]
Addressed to the extent that is possible in #5835.
